commit 7a7455aa17e109c0783d2a8ac0d1a5b344ef359b
Author: Mike Gerdts <mike.gerdts@joyent.com>
Date:   2019-05-21T18:21:29+00:00 (5 months ago)
    
    TRITON-1685 linux-prepare-image should clear /etc/machine-id
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/tools/prepare-image/linux-prepare-image b/tools/prepare-image/linux-prepare-image
index 4c5da8a..597134c 100755
--- a/tools/prepare-image/linux-prepare-image
+++ b/tools/prepare-image/linux-prepare-image
@@ -129,6 +129,12 @@ function cleanup_hostname() {
     touch /etc/hostname
 }
 
+function cleanup_machineid() {
+    if [[ -f /etc/machine-id ]]; then
+        cp /dev/null /etc/machine-id
+    fi
+}
+
 function cleanup_network_devices() {
     if [[ -f /etc/udev/rules.d/70-persistent-net.rules ]] ; then
         rm -f /etc/udev/rules.d/70-persistent-net.rules
@@ -258,6 +264,7 @@ cleanup_ssh
 cleanup_root
 cleanup_metadata
 cleanup_hostname
+cleanup_machineid
 cleanup_network_devices
 cleanup_cloud_init
 
